lib/rules/prefer-wait-for.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const { getDocsUrl } = require('../utils');
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Use `waitFor` instead of deprecated wait methods',
+      category: 'Best Practices',
+      recommended: false,
+      url: getDocsUrl('prefer-wait-for'),
+    },
+    messages: {
+      preferWaitFor: '`{{ methodName }}` is deprecated in favour of `waitFor`',
+    },
+    fixable: 'code',
+    schema: [],
+  },
+
+  create: function(context) {
+    const reportWait = node => {
+      context.report({
+        node: node,
+        messageId: 'preferWaitFor',
+        data: {
+          methodName: node.name,
+        },
+        fix: fixer => {
+          const { parent } = node;
+          const [arg] = parent.arguments;
+
+          if (arg) {
+            return fixer.replaceText(node, 'waitFor');
+          }
+          return fixer.replaceText(parent, 'waitFor(() => {})');
+        },
+      });
+    };
+
+    const reportWaitForDomChange = node => {
+      context.report({
+        node: node,
+        messageId: 'preferWaitFor',
+        data: {
+          methodName: node.name,
+        },
+        fix: fixer => {
+          const { parent } = node;
+          const [arg] = parent.arguments;
+          if (arg) {
+            return [
+              fixer.replaceText(node, 'waitFor'),
+              fixer.insertTextBefore(arg, `() => {}, `),
+            ];
+          }
+          return fixer.replaceText(parent, 'waitFor(() => {})');
+        },
+      });
+    };
+
+    return {
+      'CallExpression > Identifier[name=wait]'(node) {
```

Comment:
What if the user is using a custom `wait` util? The rule will report it even if it's not imported from Testing Library?

Reply:
Yep. I think it's too much from where it was imported? But I don't know what to do to be honest.

Reply:
Obviously it would be better to check where it's imported from, but we have been checking this in other rules without caring about it. Could be an improvement for a separated ticket for all the rules within the plugin?

Reply:
On the other hand... It would be interesting to check from where it's imported so we can also replace the import itself. Otherwise, the fix of this rule could leave a broken file.

Reply:
I would suggest keep it like this for now and wait for users to ask for it, in case it's a valid use case.

Reply:
Unless it's not too complicated to implement of course 😅